src/katello/tracer/dnf.py

```diff
@@ -0,0 +1,66 @@
+import re
+import os
+import subprocess
+from katello.tracer.SystemdDbus import SystemdDbus
+
+class DnfTracerApp:
+    def __init__(self, name, helper, type):
+        self.name = name
+        self.helper = helper
+        self.type = type
+
+
+def get_name(pid):
+    bus = SystemdDbus()
+    if pid and bus.unit_path_from_pid(pid):
+        if not bus.has_service_property_from_pid(pid, 'PAMName'):
+            Id = bus.get_unit_property_from_pid(pid, 'Id')
+            if Id and re.search("\.service$", Id):
+                return re.sub('\.service$', '', Id)
+    return None
+
+
+def run_command(command):
+    os.environ['LANG'] = 'C'
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+   
+    return_output = ''
+    return_code = process.wait()
+    
+    if return_code == 0:
+        output = process.stdout.read().decode('utf-8')
+        return_output = output.strip()
+    return return_code, return_output
+
+
+def collect_services_state():
+    command = "dnf needs-restarting"
+    retval, output = run_command(command)
+
+    lines = output.split('\n')
+    pids = [line.split(' : ')[0].strip() for line in lines if ' : ' in line]
+
+    apps = set()
+    for pid in pids:
+        service = get_name(pid)
+        if service is not None:
+            app = DnfTracerApp(service, "systemctl restart " + service, "systemd")
```

Comment:
I suppose we should keep the type "daemon" like it is currently for services:

```suggestion
            app = DnfTracerApp(service, "systemctl restart " + service, "daemon")
```

Reply:
Ah, I thought its systemd because it was already systemd in zypper.
Changed it to daemon for dnf and for deb/zypper, too,
Then we are in sync for all different pkg mangers again.